README.md

```diff
@@ -1,6 +1,59 @@
-# lit-element
+# LitElement
 
-## Base class for creating custom elements using Polymer and lit-html.
+## A simple base class for creating custom elements rendered with lit-html.
+
+LitElement uses [lit-html](https://github.com/Polymer/lit-html) to render into the
+element's Shadow DOM and [Polymer's](https://github.com/Polymer/polymer)
+PropertiesMixin to help manage element properties and attributes.
+
+LitElement uses a functional reactive pattern where any changes to element
+properties are batched and then rendered asynchronously. Users describe
+the entire rendering state declaratively, and lit-html ensures that updates
+are fast by creating the static DOM once and smartly updating only the parts of
+the DOM that change.
+
+For basic usage:
+
+  1. Create a class that extends `LitElement`.
+  1. Implement a static `properties` getter that returns the element's properties
+  (which automatically become observed attributes).
+  1. Then implement a `_render(props)` method and use the element's
+current properties (props) to return a `lit-html` template result to render
+into the element.
+
+See the [source](https://github.com/PolymerLabs/lit-element/blob/master/src/lit-element.ts#L90)
+ for detailed API info, here are some highlights:
+
+  * `_firstRendered()`: Called after the element DOM is rendered for the first time.
+
+  * `_shouldRender(props, changed, old)`: Implement to control if rendering
+  should occur when property values change or `invalidate` is called.
+  By default, this method always returns true, but this can be customized as
+  an optimization to avoid rendering work when changes occur which should not be rendered.
+
+  * `_render(props)`: Implement to describe the element's DOM using `lit-html`. Ideally,
+  the `_render` implementation is a pure function using only `props` to describe
+  the element template.
+
+  * `_didRender(props, changed, old)`: Called after element DOM has been rendered.
+  Implement to directly control rendered DOM. Typically this is not needed as `lit-html`
+  can be used in the `_render` method to set properties, attributes, and
+  event listeners. This method can, however, be useful to call methods
+  on rendered elements or to specifically react to the rendered
+  state of the DOM.
```

Comment:
This reads really confusingly to me. I think this API is necessary but is not really captured by the description here. I don't really understand what/why `props, changed, old` are.

One use case I've encountered is usually around needing to operate on the rendered DOM. Writing the `_render()` function exposes a string, but there's cases in which you need to operate on the produced DOM. For example, immediately focusing on a child element; this isn't possible from the `_render` function.

Reply:
s/changed/listOfChangedProperties maybe? And maybe show a snippet that has a simple/useful `didRender` example?